### PR TITLE
RUMM-2138 make SDK Features simple classes

### PR DESCRIPTION
**Pull request by xgouchet**
### What does this PR do?

Converts all `SdkFeature` `object`s to `class`es.

**NOTE**:  tests were not updated at this stage, this is a transitional state that will be followed by more refactoring.

### CODE REVIEW

**Review by 0xnm — APPROVED** [review]
I've added some questions/concerns, but overall lgtm and these are non-blocking.

With this change we have quite a lot of usages of `?.` operator and `null` checks, but I hope later it will go away (otherwise we need to think how to we check invariants of the SDK state - for example for `A` to work we should have `B` created, and if it is not the case, we should report this violation).

File: dd-sdk-android/src/main/kotlin/com/datadog/android/DatadogInterceptor.kt [hunk 1]
```diff
@@ -144,7 +143,8 @@ internal constructor(
 
     /** @inheritdoc */
     override fun intercept(chain: Interceptor.Chain): Response {
-        if (RumFeature.isInitialized()) {
+        val rumFeature = (Datadog.globalSDKCore as? DatadogCore)?.rumFeature
```
[the remark this answers is not in the document]
Comment (xgouchet) [reply dd-sdk-android/src/main/kotlin/com/datadog/android/DatadogInterceptor.kt:146]: A lot of the dependencies will be reversed. 
In v1 we had something like : 
```
class SomethingUsingFeatureA {
    fun foo() {
        get FeatureA.component
    }
}
```

In v2 we'll have: 

```
class SomethingUsingFeatureA(component: FeatureAComponent)

class FeatureA {
    buildSomething(): SomethingUsingFeatureA
}
```

File: dd-sdk-android/src/main/kotlin/com/datadog/android/core/internal/CoreFeature.kt [hunk 2]
```diff
@@ -466,4 +426,53 @@ internal object CoreFeature {
     }
 
     // endregion
+
+    companion object {
+        internal var processImportance: Int =
```
[the remark this answers is not in the document]
Comment (xgouchet) [reply dd-sdk-android/src/main/kotlin/com/datadog/android/core/internal/CoreFeature.kt:431]: In the end it'll be in the `SDKContext`, I kept it there to minimize the impact of the current feature

File: dd-sdk-android/src/main/kotlin/com/datadog/android/core/internal/data/upload/UploadWorker.kt [hunk 3]
```diff
@@ -35,28 +32,21 @@ internal class UploadWorker(
...
-        uploadAllBatches(
-            TracingFeature.persistenceStrategy.getReader(),
-            TracingFeature.uploader
-        )
-
-        // Upload RUM
-        uploadAllBatches(
-            RumFeature.persistenceStrategy.getReader(),
-            RumFeature.uploader
+        val globalSDKCore: DatadogCore? = (Datadog.globalSDKCore as? DatadogCore)
```
[the remark this answers is not in the document]
Comment (xgouchet) [reply dd-sdk-android/src/main/kotlin/com/datadog/android/core/internal/data/upload/UploadWorker.kt:35]: Not necessary, it'll disappear eventually

File: dd-sdk-android/src/main/kotlin/com/datadog/android/rum/RumMonitor.kt [hunk 4]
```diff
@@ -289,8 +289,11 @@ interface RumMonitor {
          * Builds a [RumMonitor] based on the current state of this Builder.
          */
         fun build(): RumMonitor {
-            val rumApplicationId = CoreFeature.rumApplicationId
-            return if (!RumFeature.isInitialized()) {
+            val datadogCore = Datadog.globalSDKCore as? DatadogCore
+            val coreFeature = datadogCore?.coreFeature
```
[the remark this answers is not in the document]
Comment (xgouchet) [reply dd-sdk-android/src/main/kotlin/com/datadog/android/rum/RumMonitor.kt:293]: In the end yes, it'll be that way

File: dd-sdk-android/src/main/kotlin/com/datadog/android/webview/DatadogEventBridge.kt [hunk 5]
```diff
@@ -129,19 +130,29 @@ internal constructor(
...
-                    timeProvider = CoreFeature.timeProvider,
-                    contextProvider = contextProvider
-                ),
-                WebViewLogEventConsumer(
-                    userLogsWriter = WebViewLogsFeature.persistenceStrategy.getWriter(),
-                    rumContextProvider = contextProvider,
-                    timeProvider = CoreFeature.timeProvider
+
+            if (coreFeature == null || webViewLogsFeature == null || webViewRumFeature == null) {
+                return NoOpWebViewEventConsumer()
```
[the remark this answers is not in the document]
Comment (xgouchet) [reply dd-sdk-android/src/main/kotlin/com/datadog/android/webview/DatadogEventBridge.kt:140]: It'll disappear before the end of the v2 refactor

**Pull request merged**